torch/csrc/autograd/engine.cpp

```diff
@@ -144,18 +144,30 @@ static thread_local std::shared_ptr<ReadyQueue> local_ready_queue = nullptr;
 // stream used to run the function OR the inputs are on different devices
 // and the function is responsible for properly acquiring them.
 //
-// Historically, the autograd engine ran all CUDA operations on their
-// device's DEFAULT stream. This meant that syncing (implicitly or
-// explicitly) with the default streams was required before and after
-// calling backward(). It also meant, however, that syncing with
-// the default streams after backward() was sufficient to ensure
-// that backward() had finished running. To preserve this historic
-// behavior the engine records "leaf streams," the streams of the
-// leaf variables, and syncs them with their device's default stream
-// at the end of backward. All other streams are already synchronized
-// to happen before at least one leaf stream (per the above), so syncing
-// the leaf streams with the default streams is sufficient to implement
-// the historic behavior.
+// The stream semantics of a backward() (or torch.autograd.grad()) call
+// with respect to surrounding ops are the same as for any other call:
+//
+// Safe
```

Comment:
nit: maybe just point at the doc note on cuda? it is much more detailed

Reply:
That's a better idea.